src/compiler/checker.ts

```diff
@@ -19073,11 +19073,22 @@ namespace ts {
                     let baseTypes = getBaseTypes(containingType as InterfaceType);
                     while (baseTypes.length) {
                         const baseType = baseTypes[0];
-                        if (modifiers & ModifierFlags.Protected &&
-                            baseType.symbol === declaration.parent.symbol) {
-                            return true;
+                        if (baseType.flags & TypeFlags.Intersection) {
+                            for (const intersectionMember of (baseType as IntersectionType).types) {
+                                if (getObjectFlags(intersectionMember) & (ObjectFlags.Class | ObjectFlags.Interface)) {
+                                    baseTypes = getBaseTypes(intersectionMember as InterfaceType);

```

Comment:
why stop at the first one?

Reply:
it can be an intersection of two types with constructors, right?

Reply:
Intersection types technically allow it, but we're using an intersection to approximate a mixin, where that won't be the case. It'll be one interface type (the original base) with one (or more) anonymous object type(s) with the mixin'd members. And we require mixin contructors have `...any[]` constructor signatures if one is present, specifically so we know they don't muck with the expected constructors implied by the base class.

Reply:
but i can have a const whose type is a mixin type.. we filter out the ones we do not like using isMixinConstructorType

Reply:
I've refactored the code a bit and explicitly handled mixins.